protocols/virtualdefund/virtualdefund.go

```diff
@@ -106,6 +107,29 @@ func NewObjective(preApprove bool,
 
 }
 
+// ConstructObjectiveFromState takes in a message and constructs an objective from it.
+// It accepts the message, myAddress, and a function to to retrieve ledgers from a store.
+func ConstructObjectiveFromState(
+	initialState state.State,
+	getChannel GetChannelByIdFunction,
+	getTwoPartyConsensusLedger GetTwoPartyConsensusLedgerFunction,
+) (Objective, error) {
+	channelId, err := initialState.ChannelId()
+	if err != nil {
+		return Objective{}, err
+	}
+	paidToBob := big.NewInt(0) // TODO how to set this properly?
```

Comment:
In standup today we agreed that this issue should be punted pending discussion #617 .

If it is the case that we need to "'query the consuming application to get/verify `paidToBob`" (rather than querying `V := getChannel(id)` then we can put a do that via the planned "objective approval system" #213 . We could put a TODO in to show that.